core-bundle/src/Security/Voter/DataContainer/AbstractDynamicPtableVoter.php

```diff
@@ -78,16 +78,33 @@ private function getParentTableAndId(int $id): array
 
         // Limit to a nesting level of 10
         $records = $this->connection->fetchAllAssociative(
-            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=?".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid", 9),
-            [$id],
+            "SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=:id".str_repeat(" UNION SELECT id, @pid:=pid AS pid, ptable FROM $table WHERE id=@pid AND ptable=:ptable", 9),
+            ['id' => $id, 'ptable' => $table],
         );
 
+        if (!$records) {
+            throw new \RuntimeException(\sprintf('Parent record of %s.%s not found', $table, $id));
```

Comment:
This should IMHO be a `LogicException` or an `InvalidArgumentException` (depending on if the `$id` argument is believed to exist in the code path calling the function or not).

Reply:
I don't think that's true. A record is missing in the database, which is a runtim exception. This is never supposed to happen because you cannot be viewing such a thing in the back end in the first place.

Reply:
But then the voter should never be executed, no? (which would be a logic error)

Reply:
its a parent record that is missing. The voter tries to vote on a record, but a parent record (which is necessary to vote) is not found at runtime.